web_external/views/manageJournal.js

```diff
@@ -9,23 +9,23 @@ var manageJournalView = View.extend({
 
     events: {
     },
-    initialize: function (subId) {
+    initialize: async function (subId) {
         var allIssues = [];
         this.render();
-        restRequest({
+        const resp = await restRequest({
             type: 'GET',
             path: 'journal'
-        }).done((resp) => {
-            for (var index in resp) {
-                restRequest({
-                    type: 'GET',
-                    path: `journal/${resp[index]._id}/issues`
-                }).done((jrnResp) => {
-                    allIssues = allIssues.concat(jrnResp);
-                    this.$('#journalListing').html(ManageJournalsEntryTemplate({ issueInfo: allIssues, parentInfo: resp }));
-                });
-            }
-        }); // End getting of OTJ Collection value setting
+        });
+
+        for (var index in resp) {
```

Comment:
@josephsnyder, @brianhelba: this is the one place where I believe my mechanical transformation from promise chains to async/await has caused a change in the semantics.

Originally, the code runs through the for loop and fires off all the requests at once; when each request resolves, its callback fires and the state change occurs. The transformed code serialized these requests in the order they come off the for loop.

@josephsnyder says he thought the old code had this new behavior. It's worth a discussion to determine whether this behavior is desired. On the one hand, the serialized behavior causes a consistent ordering of the results. On the other, it is generally speaking less performant.

With a bit of work, however, we could easily get higher performance while still maintaining a consistent order. It just depends on (1) how important consistent ordering is to the application and (2) how much benefit we stand to gain by improving the performance.

A final concern if we do go back to the high-concurrency approach is that we may in that case want to limit how many requests are in flight at once, also for performance reasons. And again, this only matters if typical usage would cause a problem with too many requests in-flight.